src/backend/distributed/commands/utility_hook.c

```diff
@@ -171,6 +171,18 @@ multi_ProcessUtility(PlannedStmt *pstmt,
 
 	parsetree = pstmt->utilityStmt;
 
+	if (IsA(parsetree, TransactionStmt))
```

Comment:
Probably not a common case, but we have this edge case, where a second BEGIN might override the settings.

It works fine, but could be an interesting regression test (just in case PG changes something and Citus crashes etc.


```SQL
SELECT current_setting('transaction_isolation');
┌─────────────────┐
│ current_setting │
├─────────────────┤
│ read committed  │
└─────────────────┘
(1 row)

 BEGIN;                                         BEGIN
Time: 0.268 ms
 BEGIN TRANSACTION ISOLATION LEVEL SERIALIZABLE;
WARNING:  there is already a transaction in progress
BEGIN
Time: 0.342 ms
SELECT current_setting('transaction_isolation');
┌─────────────────┐
│ current_setting │
├─────────────────┤
│ serializable    │
└─────────────────┘
(1 row)

Time: 0.334 ms

table distributed_table_cdtc ;
NOTICE:  issuing BEGIN TRANSACTION ISOLATION LEVEL SERIALIZABLE;SELECT assign_distributed_transaction_id(0, 2, '2022-11-10 10:00:12.64457+01');
DETAIL:  on server onderkalaci@localhost:9700 connectionId: 1
...
```